tests/ui/lint/unused-visibilities.stderr

```diff
@@ -0,0 +1,31 @@
+warning: visibility qualifiers have no effect on `const _` declarations
+  --> $DIR/unused-visibilities.rs:6:1
+   |
+LL | pub const _: () = {};
+   | ^^^ help: remove the qualifier
+   |
+note: the lint level is defined here
+  --> $DIR/unused-visibilities.rs:4:9
+   |
+LL | #![warn(unused_visibilities)]
+   |         ^^^^^^^^^^^^^^^^^^^
```

Comment:
Can we add a note explaining *why*? If you take a few seconds to interpret the code, you can see that the `const` is unnameable/unusable, so making it public is useless, but someone who got to this code "by accident" might not be able to put two and two together and just be confused. The main message is great, it says "what", but not "why", which we can put in a note. This can prevent someone from having to unnecessarily go to users.rust-lang.org :)